docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,200 @@
+S3OutputModule
+==============
+
+### About Amazon S3
+-------------------
+
+**Amazon S3 (Simple Storage Service)** is an object storage system with a web service interface to store and retrieve any amount of data at any time from anywhere on the web, offered by Amazon Web Services.
+
+### Overview
+------------
+
+S3OutputModule is to upload files/directories into an S3. Parallelism of uploading files into S3 can be achieved by partitioning the AbstractFileOutputOperator.
+But, S3OutputModule is needed to parallel uploading the blocks of a file into S3. 
+
+Below table illustrates the benefits of this module over AbstractFileOutputOperator.
+
+| S3OutputModule                                                                                                                                                                           | AbstractFileOutputOperator                                                                                                                                                                                                                           |
+|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Maximum file size for upload can be 5TB.                                                                                                                                                 | Maximum file size for upload can be 5GB.                                                                                                                                                                                                             |
+| Best fit for large files and small files.                                                                                                                                                | Best fit for small files.                                                                                                                                                                                                                            |
+| Module uses AmazonS3Client API's to upload objects into S3. Large files will upload using multipart feature and small files(Number of blocks is 1) will upload using putObject(...) API. | Operator uses Hadoop filesystems like S3AFileSystem, S3NFileSystem, etc. Consists of couple of steps to upload object into S3: (1) Write the data into the local filesystem. (2) When the stream close, filesystem uploads the local object into S3. |
```

Comment:
files(Number  of blocks is 1) => files (single block)
stream close => stream closes